File: gbcs-api/src/main/java/net/woggioni/gbcs/api/CallHandle.java
package net.woggioni.gbcs.api;

import net.woggioni.gbcs.api.event.RequestEvent;

import java.util.concurrent.CompletableFuture;

public interface CallHandle<T> {
    void postEvent(RequestEvent evt);
    CompletableFuture<T> call();
}

File: gbcs-server-memcached/src/main/kotlin/net/woggioni/gbcs/server/memcached/client/CallHandle.kt
package net.woggioni.gbcs.server.memcached.client

import java.nio.ByteBuffer
import java.util.concurrent.CompletableFuture

interface CallHandle {
    fun sendChunk(requestBodyChunk : ByteBuffer)
    fun waitForResponse() : CompletableFuture<Short>
}